### PR TITLE
SS-2: create filters on restaurants page

### PR DESCRIPTION
**Pull request by SaraSpahic**
- Price filter: Completed
- Rating filter: Problems with pagination
- Cuisine filter: Problems with request string

### CODE REVIEW

File: app/services/RestaurantService.java [hunk 1]
```diff
@@ -30,228 +30,288 @@
...
+        return true;
+    }
+
+    /**
+     * Find restaurants with filter pagination adapter.
+     *
+     * @param restaurantFilter the restaurant filter
+     * @return the pagination adapter
+     */
+    @SuppressWarnings("unchecked")
```
Comment (SaraSpahic) [inline app/services/RestaurantService.java]: I accidentally reformatted the code so now it looks that whole file is changed, but my actual changes are only in method findRestaurantsWithFilter().

File: app/controllers/RestaurantController.java [hunk 2]
```diff
@@ -81,6 +84,9 @@ public Result getAllRestaurants() {
 						.setPageSize(getQueryInt(request().getQueryString(PAGE_SIZE), DEFAULT_PAGE_SIZE))
 						.setNameFilter(request().getQueryString(NAME_FILTER))
 						.setCityFilter(!StringUtil.isNullOrEmpty(cityFilter) ? UUID.fromString(cityFilter) : null)
+                        .setPriceFilter(getQueryInt(request().getQueryString(PRICE_FILTER),0))
+						.setCuisineFilter(request().getQueryString(CUISINE_FILTER))
```
Comment (irfankrijestorac) [inline app/controllers/RestaurantController.java]: Here you should use `String[] cuisineFilter = request().queryString().get(CUISINE_FILTER);` to get list of all cousines not only first one.

File: app/models/helpers/RestaurantFilter.java [hunk 3]
```diff
@@ -8,106 +8,161 @@
...
+    /**
+     * The Rating.
+     */
+    public Integer rating;
+
+
+    /**
+     * The Cuisine.
+     */
+    public String cuisine;
```
Comment (irfankrijestorac) [inline app/models/helpers/RestaurantFilter.java]: Now we have a list here, not only one String

**Review by irfankrijestorac — CHANGES_REQUESTED** [review]

File: app/services/RestaurantService.java [hunk 4]
```diff
@@ -30,228 +30,288 @@
...
+
+        if (restaurantFilter.sortBy.equals("price")) {
+            criteria.addOrder(Order.desc("priceRange"));
+        }
+
+        criteria.addOrder(Order.asc("name"));
+
+        List<Restaurant> restaurants = criteria.list();
+
+        if (restaurantFilter.rating != null && restaurantFilter.rating != 0) {
```
Comment (irfankrijestorac) [inline app/services/RestaurantService.java]: How about get all reviews that meet filter conditions, and from reviews list get list of restaurant ids that should be returned?

**Review by irfankrijestorac — CHANGES_REQUESTED** [review]

**Pull request merged**